src/Features/CSharp/Portable/ExtractMethod/CSharpSelectionResult.cs

```diff
@@ -189,23 +189,11 @@ public SyntaxNode GetInnermostStatementContainer()
             return last.Parent.Parent;
         }
 
-        public override bool ContainsNonReturnExitPointsStatements(ImmutableArray<SyntaxNode> jumpsOutOfRegion)
-            => jumpsOutOfRegion.Any(n => n is not ReturnStatementSyntax);
+        public override bool ContainsNonReturnExitPointsStatements(ImmutableArray<SyntaxNode> exitPoints)
+            => exitPoints.Any(n => n is not ReturnStatementSyntax);
 
-        public override ImmutableArray<StatementSyntax> GetOuterReturnStatements(SyntaxNode commonRoot, ImmutableArray<SyntaxNode> jumpsOutOfRegion)
-        {
-            var container = commonRoot.GetAncestorsOrThis<SyntaxNode>().Where(a => a.IsReturnableConstruct()).FirstOrDefault();
-            if (container == null)
-                return [];
-
-            // now filter return statements to only include the one under outmost container
-            return jumpsOutOfRegion
-                .OfType<ReturnStatementSyntax>()
-                .Select(returnStatement => (returnStatement, container: returnStatement.GetAncestors<SyntaxNode>().Where(a => a.IsReturnableConstruct()).FirstOrDefault()))
-                .Where(p => p.container == container)
-                .SelectAsArray(p => p.returnStatement)
-                .CastArray<StatementSyntax>();
-        }
+        public override ImmutableArray<StatementSyntax> GetOuterReturnStatements(SyntaxNode commonRoot, ImmutableArray<SyntaxNode> exitPoints)
+            => exitPoints.OfType<ReturnStatementSyntax>().ToImmutableArray().CastArray<StatementSyntax>();
```

Comment:
so the prior logic was just completely unneccessary.  it seems like it was written thinking that it would be passed *all* return-like statements in a span (for example, a return-statement inside an inner lambda), versus *only* the return statements *out of that span*.  As such, it did all this complex work to filter down to only the actual return-statements returning from the span.  Except that's what 'exit points' already gives you (plus things like goto/break/continue).  So this can just be written to filter down to ReturnStatements and be done.